### PR TITLE
Ensure MethodToStateMachineRewriter.VisitScope properly rewrites local symbols referenced by the node.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems?id=401335.

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
Is Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ImplementInterface.ImplementInterfaceTests.TestQualifiedNameExplicitInterface flaky?
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_unit32_prtest/4047

**Comment by AlekseyTs** [comment]
Test windows_release_unit32_prtest please.

**Comment by AlekseyTs** [comment]
Test windows_debug_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
Test ubuntu_16_debug_prtest please

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternSwitchTests.cs [hunk 1]
```diff
@@ -2541,6 +2541,127 @@ static bool TakeOutVar(out int x)
         }
 
         [Fact]
+        [WorkItem(16066, "https://github.com/dotnet/roslyn/issues/16066")]
+        [WorkItem(16066, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=401335")]
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternSwitchTests.cs]: >16066 [](start = 18, length = 5)

Nit: workItem ID is not the one from VSO.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
Test windows_debug_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
Test windows_release_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by cston** [comment]
LGTM

**Pull request merged**